### PR TITLE
doc: redirects: review and update redirects

### PR DESCRIPTION
**Pull request by greg-fer**
Checked existing docs vs v1.8.0 for existing redirects. Sections checked: Introduction, Getting started guides, Installation, Create an application, Configuration and building, Test optimization, Device config guides, Security, Releases and maturity, Dev model. Added missing redirects.
Reported by @b-gent.

### CODE REVIEW

File: doc/_utils/redirects.py [hunk 1]
```diff
@@ -292,7 +301,6 @@
     ("device_guides/working_with_pmic/npm1300/developing","device_guides/pmic/npm1300"),
     ("device_guides/working_with_pmic/npm1300/features","device_guides/pmic/npm1300"),
     ("device_guides/working_with_pmic/npm1300/gs","device_guides/pmic/npm1300"),
-    ("known_issues","releases_and_maturity/known_issues"),
```
Comment (greg-fer) [inline doc/_utils/redirects.py:295]: Moved to line 93.

File: doc/_utils/redirects.py [hunk 2]
```diff
@@ -77,12 +77,20 @@
...
     ("dm_adding_code", "releases_and_maturity/developing/adding_code"),
     ("developing/adding_code", "releases_and_maturity/developing/adding_code"),
+    ("releases_and_maturity/developing/adding_code", "dev_model_and_contributions/adding_code"),
     ("dm_code_base", "releases_and_maturity/developing/code_base"),
     ("developing/code_base", "releases_and_maturity/developing/code_base"),
-    ("dm_managing_code", "releases_and_maturity/developing/managing_code"),
+    ("releases_and_maturity/developing/code_base", "dev_model_and_contributions/code_base"),
+    ("dm_managing_code", "developing/managing_code"),
     ("developing/managing_code", "releases_and_maturity/developing/managing_code"),
+    ("releases_and_maturity/developing/managing_code", "dev_model_and_contributions/managing_code"),
```
Comment (greg-fer) [inline doc/_utils/redirects.py:86]: 82 duplicated in 85

File: doc/_utils/redirects.py [hunk 3]
```diff
@@ -77,12 +77,20 @@
     ("dev_model", "releases_and_maturity/dev_model"),
     ("dm_adding_code", "releases_and_maturity/developing/adding_code"),
     ("developing/adding_code", "releases_and_maturity/developing/adding_code"),
+    ("releases_and_maturity/developing/adding_code", "dev_model_and_contributions/adding_code"),
     ("dm_code_base", "releases_and_maturity/developing/code_base"),
     ("developing/code_base", "releases_and_maturity/developing/code_base"),
-    ("dm_managing_code", "releases_and_maturity/developing/managing_code"),
+    ("releases_and_maturity/developing/code_base", "dev_model_and_contributions/code_base"),
+    ("dm_managing_code", "developing/managing_code"),
     ("developing/managing_code", "releases_and_maturity/developing/managing_code"),
```
Comment (greg-fer) [inline doc/_utils/redirects.py:85]: The removed line skipped some structure changes.

**Review by wiba-nordic — APPROVED** [review]
Everything looks good to me.

**Review by umapraseeda — APPROVED** [review]

**Review by b-gent — APPROVED** [review]

**Pull request merged**